tsl/src/bgw_policy/compression_api.c

```diff
@@ -37,39 +31,24 @@
 #define DEFAULT_RETRY_PERIOD                                                                       \
 	DatumGetIntervalP(DirectFunctionCall3(interval_in, CStringGetDatum("1 hour"), InvalidOid, -1))
 
-#define POLICY_COMPRESSION_PROC_NAME "policy_compression"
-#define POLICY_RECOMPRESSION_PROC_NAME "policy_recompression"
-#define CONFIG_KEY_HYPERTABLE_ID "hypertable_id"
-#define CONFIG_KEY_COMPRESS_AFTER "compress_after"
-#define CONFIG_KEY_RECOMPRESS_AFTER "recompress_after"
-#define CONFIG_KEY_RECOMPRESS "recompress"
-#define CONFIG_KEY_MAXCHUNKS_TO_COMPRESS "maxchunks_to_compress"
```

Comment:
`compression_api.c`, `reorder_api.c` and `retention_api.c` all have their own `#define` for `CONFIG_KEY_HYPERTABLE_ID`.

When moving all policy json identifiers to the same file, it's more confusing to merge any keys that happen to be identical, rather than have the identifiers clearly laid out per policy.

This was a tradeoff between code readability versus history of commits. It is a kind of refactoring. Imagine `CONFIG_KEY_HYPERTABLE_ID` being something other than `"hypertable_id"` for some policies. Not having clear per-policy keys is very error prone and should be corrected somehow.

Reply:
> compression_api.c, reorder_api.c and retention_api.c all have their own #define for CONFIG_KEY_HYPERTABLE_ID.
>
> This was a tradeoff between code readability versus history of commits. It is a kind of refactoring. Imagine `CONFIG_KEY_HYPERTABLE_ID` being something other than `"hypertable_id"` for some policies.

That is possible, but it it's a little premature since they are identical right now (and I suspect that they actually need to be the same in both the old and the new interface to avoid confusion).

It is better to change the names when and if there is actually a need for this separation rather than add complications at this point when it is not needed.

> Not having clear per-policy keys is very error prone and should be corrected somehow.

Not sure how this can be error prone. Could you elaborate?

Reply:
Let's say for example that somebody wants to change the value of:
```
CONFIG_KEY_HYPERTABLE_ID "hypertable_id"
```
for `compression_api.c`.

They might do something like:
```
CONFIG_KEY_HYPERTABLE_ID "new_hypertable_id"
CONFIG_KEY_LEGACY_HYPERTABLE_ID "hypertable_id"
```

However, unbeknownst to them, they modified `CONFIG_KEY_HYPERTABLE_ID` for `retention_api.c` as well. This is going to be getting more confusing and error prone the more we add new policy functionality.